server_action_logging/models/ir_actions_server.py

```diff
@@ -0,0 +1,66 @@
+# © 2024 Camptocamp (https://www.camptocamp.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+import logging
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_LOG_CONFIGURATION = [
+    "odoo.http.rpc.request:INFO",
+    "odoo.http.rpc.response:INFO",
+    ":INFO",
+]
+
+
+class IrActionsServer(models.Model):
+
+    _inherit = "ir.actions.server"
+
+    enable_sql_debug = fields.Boolean(
+        string="Enable SQL Debug", help="Enable SQL Debug for this action", default=True
+    )
+
+    def run(self):
+        log_handlers = DEFAULT_LOG_CONFIGURATION
+        res = False
+        for action in self:
+            if action.enable_sql_debug:
+                self._update_logger_level(log_handlers)
+            start_time = fields.Datetime.now()
+            _logger.info(
+                "Running action with id %s and name %s",
+                action.id,
+                action.name,
+            )
+            try:
+                res = super(IrActionsServer, action).run()
+            finally:
+                _logger.info(
+                    "Action with id %s and name %s took %s seconds",
+                    action.id,
+                    action.name,
+                    fields.Datetime.now() - start_time,
```

Comment:
Please use `timedelta.total_seconds()` method if you want to display duration in seconds.